src/mds/CDir.cc

```diff
@@ -2455,25 +2463,23 @@ void CDir::_omap_commit_ops(int r, int op_prio, int64_t metapool, version_t vers
 
   int count = 0;
   for (auto &key : stales) {
-    unsigned size = key.length() + sizeof(__u32);
-    if (write_size + size > max_write_size)
```

Comment:
This will mean the MDS will routinely exceed the size of the config `mds_dir_max_commit_size`. Why not:

`if (write_size > 0 && write_size + size > max_write_size)`

Reply:
This also looks good to me and much simpler. 

Usually for the `stales` and `to_remove` cases it should be okay, the size of metadata won't be too big for a single item. But not for the `to_set`, which will encode the inode base.

I will revise them all. Thanks @batrick